static_analyzer/clustering/engine.py

```
@@ -0,0 +1,242 @@
+def _try_all_algorithms(graph: nx.DiGraph, min_cluster_size: int, total_nodes: int) -> list[Candidate]:
+    """Run Leiden and return a single scored candidate.
+
+    Returned as a list so the cross-level pooling in ``cluster_graph`` stays uniform.
+    """
+    candidates: list[Candidate] = []
+    try:
+        communities = _cluster_with_algorithm(graph, "leiden")
+        score = _score_clustering(communities, min_cluster_size, total_nodes)
+        candidates.append((communities, "leiden", score))
+        logger.debug(f"leiden: score={score:.3f}, clusters={len(communities)}")
+    except Exception as e:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)